src/averageAges.js

```diff
@@ -37,7 +43,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(
+      person => person.sex === 'f'
+      && people.some(p => p.mother === person.name)
+    )
+    : people.filter(person => person.sex === 'f');
+
+  const ages = women.map(person => person.died - person.born);
+  const totalAge = ages.reduce((prev, age) => prev + age, 0);
+  const averageAge = totalAge / women.length;
```

Comment:
this logic is the same as in the previous function. Make a separate function for it, to avoid duplication